ert_shared/ensemble_evaluator/dispatch.py

```diff
@@ -41,27 +48,86 @@ async def join(self):
 
 
 class Dispatcher:
-    def __init__(self, batcher=None):
-        self.__LOOKUP_MAP = defaultdict(list)
+    def __init__(self, ensemble, evaluator_callback, batcher=None):
+        self._LOOKUP_MAP = defaultdict(list)
         self._batcher: Optional[Batcher] = batcher
-
-    def register_event_handler(self, event_types, batching=False):
-        def decorator(function):
-            nonlocal event_types, batching
-            if not isinstance(event_types, set):
-                event_types = set({event_types})
-            for event_type in event_types:
-                self.__LOOKUP_MAP[event_type].append((function, batching))
-
-            def wrapper(*args, **kwargs):
-                return function(*args, **kwargs)
-
-            return wrapper
-
-        return decorator
+        self._ensemble = ensemble
+        self._evaluator_callback = evaluator_callback
+        self._register_event_handlers()
+
+    def register_event_handler(self, event_types, function, batching=False):
```

Comment:
Are there cases were `batch=False` is used?

Reply:
In the tests

Reply:
Minor thing, I guess, but if it is only used for testing, should we not be patching/mocking somehow instead?